### PR TITLE
Add support for nested transactions with ActiveRecord

### PR DESCRIPTION
**Pull request by EPecherkin**
### Review

- [ ] Document code according to [Getting Started with Yard](http://www.rubydoc.info/gems/yard/file/docs/GettingStarted.md).
- [ ] All tests are passing.
- [ ] Test manually.
- [ ] Get approval.

### Pre-merge checklist

- [ ] The PR relates to a single subject with a clear title and description in grammatically correct, complete sentences.
- [ ] Verify that feature branch is up-to-date with `master` (if not - rebase it).
- [ ] Double check the quality of [commit messages](http://chris.beams.io/posts/git-commit/).
- [ ] Squash related commits together.

### CODE REVIEW

File: lib/granite/action/transaction_manager.rb [hunk 1]
```diff
@@ -9,11 +9,14 @@ class << self
...
         # @return [Object] result of a block
         def transaction(&block)
-          if in_a_transaction?
-            yield
-          else
-            wrap_in_transaction_with_callbacks(&block)
-          end
+          callback_listeners << []
+
+          self.transactions_in_progress += 1
```
Comment (pluff) [inline lib/granite/action/transaction_manager.rb]: please extract callbacks tree logic to separate class.
```
TrasactionCallbacksTree.new

tree.child #marks the start of new transaction
tree.close_current(result) # marks finish of new transaction
tree.after_commit_callbacks # returns a list of after commit callbacks to be run
tree.transaction_level # returns depth of current transaction
```

**Comment by eyusupov** [comment]
@toptal-bot run backend specs

**Comment by eyusupov** [comment]
@toptal-bot run backend specs

File: spec/lib/granite/action/transaction_manager/transactions_stack_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  it 'fails to add callback' do
+    expect { subject.add_callback(callback1) }.to raise_error(RuntimeError, 'Start a transaction before you add callbacks on it')
+  end
+
+  context 'when transaction is opened' do
+    before { subject.start_new! }
+
+    it { expect(subject.depth).to eq(1) }
+
+    it 'manges callbacks' do
```
Comment (jaimerson) [inline spec/lib/granite/action/transaction_manager/transactions_stack_spec.rb]: manages

**Pull request merged**